node-repository/src/test/java/com/yahoo/vespa/hosted/provision/provisioning/ProvisioningTest.java

```diff
@@ -156,6 +157,33 @@ public void nodeVersionIsReturnedIfSet() {
         assertEquals(Version.fromString("1.2.3"), host1.version().get());
     }
 
+    @Test
+    public void dockerImageRepoIsReturnedIfSet() {
+        ProvisioningTester tester = new ProvisioningTester.Builder().zone(new Zone(Environment.dev, RegionName.from("us-east"))).build();
+
+        tester.makeReadyNodes(4, defaultResources, NodeType.host, 1);
+        tester.prepareAndActivateInfraApplication(tester.makeApplicationId(), NodeType.host);
+
+        // deploy
+        ApplicationId application1 = tester.makeApplicationId();
+        SystemState state1 = prepare(application1, tester, 1, 1, 1, 1, defaultResources, "1.2.3");
+        String dockerImageRepo = "docker.domain.tld/my/image";
+        prepare(application1, tester, 1, 1, 1 , 1 , false, defaultResources, "1.2.3", Optional.of(dockerImageRepo));
+        tester.activate(application1, state1.allHosts);
+
+        HostSpec host1 = state1.container0.iterator().next();
+        assertFalse(host1.version().isPresent());
+        Node node1 = tester.nodeRepository().getNode(host1.hostname()).get();
+        tester.nodeRepository().write(node1.with(node1.status().withDockerImage(DockerImage.fromString(dockerImageRepo))), () -> {});
```

Comment:
This actually shouldn't be allowed, current docker image should always include the version as well.

Reply:
test code updated